### PR TITLE
feat($theme-default): smooth scroll (close #567)

### LINKED ISSUE #567: Smooth scrolling when clicking header links
Contribution welcome!

### PR DESCRIPTION
**Pull request by meteorlxy**
**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

smooth-scroll

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [x] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [x] A convincing reason for adding this feature: close #567
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

The reason of changing `throttle` to `debounce` is to avoid [this issue](https://github.com/vuepress/vuepress-plugin-smooth-scroll/issues/1).

### CODE REVIEW

**Comment by ulivz** [comment]
Good job, but I found a little frame dropping in my computer, I don't know what's going on on your machine and if there's any frame dropping. 😳

**Comment by meteorlxy** [comment]
What about an option of themeConfig to enable / disable smooth-scroll ?

File: packages/docs/docs/.vuepress/config.js [hunk 1]
```diff
@@ -35,6 +35,7 @@ module.exports = ctx => ({
       apiKey: '3a539aab83105f01761a137c61004d85',
       indexName: 'vuepress'
     }) : null,
+    smoothScroll: true,
```
Comment (meteorlxy) [inline packages/docs/docs/.vuepress/config.js]: It's TBD that we should enable `smoothScroll` for official docs site. cc @ulivz

Comment (ulivz) [reply packages/docs/docs/.vuepress/config.js]: I support trying out new features in our official docs.

**Pull request merged**